yt/frontends/chimera/data_structures.py

```diff
@@ -0,0 +1,295 @@
+"""
+Chimera data structures
+
+
+
+"""
+
+import os
+
+import numpy as np
+
+from yt.data_objects.index_subobjects.unstructured_mesh import SemiStructuredMesh
+from yt.data_objects.static_output import Dataset
+from yt.geometry.geometry_handler import YTDataChunk
+from yt.geometry.unstructured_mesh_handler import UnstructuredIndex
+from yt.utilities.file_handler import HDF5FileHandler
+from yt.utilities.io_handler import io_registry
+from yt.utilities.on_demand_imports import _h5py as h5py
+
+from .fields import ChimeraFieldInfo
+
+
+class ChimeraMesh(SemiStructuredMesh):
+
+    _index_offset = 0
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+
+def _find_files(
+    filename_c,
+):  # Returns a list of all files that share a frame number with the input
+    dirname, file = os.path.split(filename_c)
+    dir_contents = os.listdir(dirname)
+    frames = [f for f in dir_contents if file[:-7] in f]
+    directory = os.path.dirname(filename_c)
```

Comment:
good call, I'll add that `os.path.splitext` is probably what we should use instead of selecting chars manually as `file[:-7]`

Reply:
Removing `directory` and using `dirname` instead sounds sensible. As for using `splitext`, it's my understanding that it only separates the base name and file extension; what `file[:-7]` is doing is also removing the section of the file name that specifies the grid number. That way, the code will match all grid files that correspond to the same time-step (the timestep is specified earlier in the file name). If there's a more elegant way to do this with `splitext` or otherwise, I'm open to it.

Reply:
I added a suggestion separate from this thread